GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -124,39 +124,112 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
             //
             Dictionary<string, string> requiredSettings = new Dictionary<string, string>
             {
+                 // When running 'git am' it will remove the CRs from the patch file by default. This causes the patch to fail to apply because the
+                 // file that is getting the patch applied will still have the CRs. There is a --keep-cr option that you can pass the 'git am' command
+                 // but since we always want to keep CRs it is better to just set the config setting to always keep them so the user doesn't have to
+                 // remember to pass the flag. https://mseng.visualstudio.com/VSOnline/_git/GVFS/pullrequest/244891?_a=overview.
                 { "am.keepcr", "true" },
+
+                // Update git settings to enable optimizations in git 2.20
+                // Set 'checkout.optimizeNewBranch=true' to enable optimized 'checkout -b'
                 { "checkout.optimizenewbranch", "true" },
+
+                // Needed to prevent line ending conversion. Prevents git from auto-replacing line endings with Crlf.
                 { "core.autocrlf", "false" },
+
+                // commit graph. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/299081
                 { "core.commitGraph", "true" },
+
+                // Perf - Git for Windows uses this to bulk-read and cache lstat data of entire directories (instead of doing lstat file by file).
                 { "core.fscache", "true" },
+
+                // Turns on all special gvfs logic. https://github.com/microsoft/scalar/blob/935c50110b31c440aae8e1cdc4ffea72bc3be01f/Scalar/CommandLine/ScalarVerb.cs#L35
                 { "core.gvfs", "true" },
+
+                // Use 'multi-pack-index' builtin instead of 'midx' to match upstream implementation
                 { "core.multiPackIndex", "true" },
+
+                // Perf - Enable parallel index preload for operations like git diff
                 { "core.preloadIndex", "true" },
                 { "core.safecrlf", "false" },

```

Comment:
Details on this setting (https://git-scm.com/docs/git-config):

> If true, makes Git check if converting CRLF is reversible when end-of-line conversion is active. Git will verify if a command modifies a file in the work tree either directly or indirectly. For example, committing a file followed by checking out the same file should yield the original file in the work tree. If this is not the case for the current setting of core.autocrlf, Git will reject the file. The variable can be set to "warn", in which case Git will only warn about an irreversible conversion but continue the operation.

My guess is that because VFS4G never wants git to adjust line endings, we're explicitly setting `core.safecrlf` to false.

Reply:
[updated](https://github.com/microsoft/VFSForGit/pull/1614/commits/a39ea1d2cf052db520e9440893b98ae3948ee1a6)